### ISSUE TITLE
Clarify that we expect models without softmax activation at the end

### ISSUE THREAD

**Issue by cofri** [deel-ai/oodeel#94]
Some methods can raise an error which is difficult to understand when a model with softmax is used. For example, VIM tries to get weights of the last layer. If last layer is Softmax, an error is raised:
https://github.com/deel-ai/oodeel/blob/d07f64c2e49acb292e37ca302aa7d14236344de4/oodeel/methods/vim.py#L102

Suggestions:
- clarify documentation/docstrings/etc. to force users to end their models with logits
- OR ensure that models used do not end with activation (error assertion)
- OR automatically remove last activation of model when `fit()`.
